src/main/java/org/openpnp/spi/PnpJobProcessor.java

```diff
@@ -1,4 +1,13 @@
 package org.openpnp.spi;
 
+import java.util.List;
+
+import org.openpnp.machine.reference.ReferencePnpJobProcessor.JobPlacement;
```

Comment:
This is a use of a concrete implementation class in a SPI class, which is not allowed. The solution is to just move JobPlacement class into the PnpJobProcessor interface, like this:

```
package org.openpnp.spi;

import java.util.List;

import org.openpnp.model.BoardLocation;
import org.openpnp.model.LengthUnit;
import org.openpnp.model.Placement;

public interface PnpJobProcessor extends JobProcessor {

    public List<JobPlacement> getJobPlacementsById(String id);

    public List<JobPlacement> getJobPlacementsById(String id, String status);

    public static class JobPlacement {
        public enum Status {
            Pending,
            Processing,
            Skipped,
            Complete
        }

        public final BoardLocation boardLocation;
        public final Placement placement;
        public Status status = Status.Pending;

        public JobPlacement(BoardLocation boardLocation, Placement placement) {
            this.boardLocation = boardLocation;
            this.placement = placement;
        }

        public double getPartHeight() {
            return placement.getPart().getHeight().convertToUnits(LengthUnit.Millimeters)
                    .getValue();
        }

        @Override
        public String toString() {
            return placement.getId();
        }
    }
}
```